node/services/service.go

```diff
@@ -229,9 +230,23 @@ func FullAvailability(cfg Config) func(
 }
 
 // CacheAvailability wraps either Full or Light availability with a cache for result sampling.
-func CacheAvailability[A share.Availability](lc fx.Lifecycle, ds datastore.Batching, avail A) share.Availability {
+func CacheAvailability[A share.Availability](
+	ctx context.Context,
+	lc fx.Lifecycle,
+	ds datastore.Batching,
+	fservice fraud.Service,
+	avail A,
+) share.Availability {
 	ca := share.NewCacheAvailability(avail, ds)
 	lc.Append(fx.Hook{
+		OnStart: func(context.Context) error {
+			switch any(avail).(type) {
+			case *share.LightAvailability:
+				fservice.Sync(fxutil.WithLifecycle(ctx, lc))
```

Comment:
I think it would be cleaner for `LightAvailability` to handle this in its own lifecycle in `service.go`